File: docs/validation/cpp/app/src/scenarios/av05_corrupt_scalar.cpp
// AV-05: Reader handles corrupt required attribute (Build_Plate_X_Dimension) gracefully
#include "av05_corrupt_scalar.hpp"
#include "machine_config/machine_config.hpp"

namespace av05 {

using namespace machine_config;

scenarios::Result run(const std::filesystem::path& fixturesDir, const std::filesystem::path&) {
    auto fixture = scenarios::avFixture(fixturesDir, "corrupt_scalar.h5");
    try {
        MachineConfigReader{fixture}.parse();
        return {false, "no error raised for corrupt Build_Plate_X_Dimension"};
    } catch (const std::exception& e) {
        return {true, std::string("error raised for corrupt scalar: ") + e.what()};
    }
}

} // namespace av05

File: docs/validation/rust/app/src/scenarios/av05_corrupt_scalar.rs
// AV-05: Reader handles corrupt required attribute (Build_Plate_X_Dimension) gracefully
use super::common::av_fixture;
use machine_config::{MachineConfigError, MachineConfigReader};
use std::path::Path;

pub fn run(fixtures_dir: &Path, _real_dir: &Path) -> (bool, String) {
    let fixture = av_fixture(fixtures_dir, "corrupt_scalar.h5");
    match MachineConfigReader::open(&fixture).and_then(|r| r.parse()) {
        Ok(_) => (false, "no error raised for corrupt Build_Plate_X_Dimension".to_string()),
        Err(e @ MachineConfigError::Parse(_)) => {
            (true, format!("Parse error raised for corrupt scalar: {e}"))
        }
        Err(e) => (false, format!("unexpected error variant: {e}")),
    }
}
